RPGManager_development/test/unit_test/character_sheet_unit_tests/ArmorsUnitTests.java

```diff
@@ -0,0 +1,150 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.security.InvalidParameterException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.Armors;
+import rpg_database.character_sheet.BaseClasses;
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+
+public class ArmorsUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetArmorsMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Armors");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, malformedInput);
+	}
+
+	// default Armors for BaseClasses
+
+	@Test
+	public void testGetDefaultArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.HEAVY_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForRogue() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.ROGUE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForMage() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.MAGE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.ROBE, actualValue);
+	}
+
+	// default ArmorRating for BaseClasses
+
+	@Test
+	public void testGetDefaultArmorRating() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		int actualArmorValue = characterSheet.getData(Fields.ARMOR_RATING);
+		assertEquals(0, actualArmorValue);

```

Comment:
Nekem az első megoldás tetszik a legjobban, ezért arra írtam át. Ha a továbbiakban több if else ág is születne, akkor majd újra gondoljuk, hogyan lehetne átdolgozni őket.